### PR TITLE
Spark: Handle CaseSenstivity in runtimeFiltering of Spark

### LINKED ISSUE #3978: [Spark] Handle caseSensitivity of runtimeFilters in runtimeFiltering Spark
Presently a query like this : 

``` SQL
SELECT f.* FROM tableName f JOIN dim d ON f.Id = d.id AND f.Data = d.data AND d.date = DATE '1970-01-02'
```
fails with
>  org.apache.iceberg.exceptions.ValidationException: Cannot find field 'Id' in struct: struct<1: id: optional long, 2: data: optional string, 3: date: optional date, 4: ts: optional timestamptz>

AFAIK this happens because while creating inclusiveExpr from RuntimeFilters passed from Spark, it doesn't takes cares of case sensitivity : [CodePointer](https://github.com/apache/iceberg/blob/master/spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java#L171)

```java
Expression inclusiveExpr = Projections.inclusive(spec).project(runtimeFilterExpr);
```
As per my understanding we should pass ``Projections.inclusive(spec, caseSensitive())`` to make sure iceberg also interprets it in the same way as spark does. Post adding this check the above UT succeeds.

### PR DESCRIPTION
**Pull request by singhpk234**
Closes #3978

### CODE REVIEW

File: spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestRuntimeFiltering.java [hunk 1]
```diff
@@ -177,13 +177,21 @@ public void testMultipleRuntimeFilters() throws NoSuchTableException {
         "SELECT f.* FROM %s f JOIN dim d ON f.id = d.id AND f.data = d.data AND d.date = DATE '1970-01-02'",
         tableName);
 
+    String caseInsensitiveQuery = String.format(
+        "SELECT f.* FROM %s f JOIN dim d ON f.Id = d.id AND f.Data = d.data AND d.date = DATE '1970-01-02'",
```
Comment (kbendick) [inline spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestRuntimeFiltering.java:181]: Can you please go all out in making this case insensitivity check a bit more clear?

Namely, finding the characters that were different from the above wasn't that quickly obvious.

I would say go wild and make as much of it as you can have funky non-standard casing. Like `select f.* from %s F join dim d ON f.Id = d.iD and f.DaTa = d.dAtA and d.dAtE = date '1970-01--02'`

**Review by kbendick — COMMENTED** [review]
Thanks for this @singhpk234!

Left a small note to hopefully make the case sensitivity check more clear, and then I'm also wondering if this should be its own named test instead of sticking it into `testMultipleRuntimeFilters`.

**Review by jackye1995 — APPROVED** [review]
nice catch! looks good to me, as long as Kyle's comment is addressed

**Review by kbendick — APPROVED** [review]
Looks good to me. Thanks @singhpk234!

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @singhpk234. Good catch!

**Comment by rdblue** [comment]
@jackye1995 should this be in the 0.13.0 release?

**Pull request merged**